src/hyperlink/_url.py

```diff
@@ -1095,22 +1170,19 @@ def from_text(cls, text):
             look like URLs are still valid URLs. As a result, this
             method only raises :class:`URLParseError` on invalid port
             and IPv6 values in the host portion of the URL.
-
         """
         um = _URL_RE.match(_textcheck('text', text))
-        try:
-            gs = um.groupdict()
-        except AttributeError:
+        if um is None:
             raise URLParseError('could not parse url: %r' % text)
+        gs = um.groupdict()
```

Comment:
If `um` isn't `None`, it shouldn't raise `AttributeError` here, so this now just checks if `um` is `None`.  That avoids some casting.